### DISCUSSION TITLE
What is the difference between Flex and Pack layouts?

### DISCUSSION THREAD

**Discussion by myyrakle** [fltk-rs/fltk-rs#1369] [Q&A] [answered]
https://fltk-rs.github.io/fltk-book/Layouts.html

I'm looking at that guide document, but I can't tell the exact difference just by looking at this document.

It seems to be the same in that it automatically aligns and arranges child components. What part is different?

**Comment by MoAlyousef** [comment, marked as answer]
They’re similar but have important differences:

If we take a column Flex vs a vertical pack, the pack needs to know the height of each child, flexes don’t. Packs also resize themselves when adding a new widget, while flexes don’t (unless you specify ut by using the Flex::set_size method. Also packs don’t resize their content when they’re resized. 
That makes packs more suitable to creating a scrollable list of items for example. 
Flexes are more suitable for general flexible layouts.

**Reply by myyrakle** [reply]
Thanks for your kind reply!
